### ISSUE TITLE
using ASTRA via tomopy on Windows (conda, compilation)

### ISSUE THREAD

**Issue by oachk** [tomopy/tomopy#364]
Hi!
I also have an issue reproducing examples that call ASTRAs CPU functions. 
If I run the example suggested for an ASTRA reconstruction via CPU:

```
import tomopy
import pylab

def main():
    obj = tomopy.shepp3d()
    # Generate an object.
        
    pylab.imshow(obj[64], cmap='gray')
    pylab.colorbar()
    pylab.show()

    ang = tomopy.angles(180)
    # Generate uniformly spaced tilt angles.
    sim = tomopy.project(obj, ang)
    rec = tomopy.recon(sim, ang, algorithm=tomopy.astra, 
                       options={'method':'SART', 
                                'num_iter':10*180,
                                'proj_type':'linear',
                                'extra_options':{'MinConstraint':0}
                                })

    # Show 64th slice of the reconstructed object.
    pylab.imshow(rec[64], cmap='gray')
    pylab.colorbar()
    pylab.show()

if __name__ =='__main__':
    
    main()
```
I get the following error:
```
runfile('C:/Users/Hanna/Documents/PythonScripts/main_test_tomopy_astra.py', wdir='C:/Users/Hanna/Documents/PythonScripts')
Traceback (most recent call last):

  File "<ipython-input-2-62983392d139>", line 1, in <module>
    runfile('C:/Users/Hanna/Documents/PythonScripts/main_test_tomopy_astra.py', wdir='C:/Users/Hanna/Documents/PythonScripts')

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 668, in runfile
    execfile(filename, namespace)

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/Hanna/Documents/PythonScripts/main_test_tomopy_astra.py", line 7, in <module>
    import tomopy

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\tomopy\__init__.py", line 57, in <module>
    from tomopy.misc.phantom import *

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\tomopy\misc\phantom.py", line 57, in <module>
    import skimage

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\__init__.py", line 177, in <module>
    from .data import data_dir

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\data\__init__.py", line 15, in <module>
    from ..io import imread, use_plugin

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\io\__init__.py", line 7, in <module>
    from .manage_plugins import *

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\io\manage_plugins.py", line 28, in <module>
    from .collection import imread_collection_wrapper

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\io\collection.py", line 14, in <module>
    from ..external.tifffile import TiffFile

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\external\tifffile\__init__.py", line 1, in <module>
    from .tifffile import imsave, imread, imshow, TiffFile, TiffWriter, TiffSequence

  File "C:\Users\Hanna\Miniconda3\lib\site-packages\skimage\external\tifffile\tifffile.py", line 293, in <module>
    from . import _tifffile

RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
```
I'm working on Windows (Intel GPU) in a conda environment equipped with python 3.6, tomopy and the astra-toolbox, see the following list of packages:

```
(tomoastra) C:\Users\Hanna>conda list
# packages in environment at C:\Users\Hanna\Miniconda3\envs\tomoastra:
#
# Name                    Version                   Build  Channel
astra-toolbox             1.8.3            py36h8016247_0    astra-toolbox
blas                      1.0                         mkl
ca-certificates           2018.03.07                    0
certifi                   2018.10.15               py36_0
cftime                    1.0.2.1         py36h452e1ab_1000    conda-forge
cloudpickle               0.6.1                      py_0    conda-forge
curl                      7.49.1                   vc14_2  [vc14]  conda-forge
cycler                    0.10.0                     py_1    conda-forge
cytoolz                   0.9.0.1         py36hfa6e2cd_1001    conda-forge
dask-core                 0.20.0                     py_0    conda-forge
decorator                 4.3.0                      py_0    conda-forge
dxchange                  0.1.4                    py36_0    conda-forge
dxfile                    0.5                      py36_0    conda-forge
edffile                   5.0.0                    py36_0    conda-forge
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
h5py                      2.8.0            py36hfa6e2cd_0    conda-forge
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
imageio                   2.4.1                 py36_1000    conda-forge
intel-openmp              2019.0                      118
jpeg                      9b                       vc14_2  [vc14]  conda-forge
kiwisolver                1.0.1           py36he980bc4_1002    conda-forge
libastra                  1.8.3                h49425a5_0    astra-toolbox
libnetcdf                 4.6.1                    vc14_2  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
matplotlib                2.2.2                    py36_1    conda-forge
mkl                       2019.0                      118
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
netcdf4                   1.4.0                    py36_0    conda-forge
networkx                  2.2                        py_1    conda-forge
numexpr                   2.6.8            py36h9ef55f4_0
numpy                     1.13.3           py36h5c71026_4
numpy-base                1.15.4           py36h8128ebf_0
olefile                   0.46                       py_0    conda-forge
openssl                   1.0.2p               hfa6e2cd_0
pillow                    5.2.0                    py36_0    conda-forge
pip                       18.1                     py36_0
pyparsing                 2.3.0                      py_0    conda-forge
pyqt                      5.6.0            py36h764d66f_7    conda-forge
pyreadline                2.1                   py36_1000    conda-forge
python                    3.6.6                he025d50_0    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pytz                      2018.7                     py_0    conda-forge
pywavelets                1.0.1           py36h452e1ab_1000    conda-forge
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
scikit-image              0.14.1          py36h6538335_1001    conda-forge
scipy                     1.1.0            py36hc28095f_0
setuptools                40.5.0                   py36_0
sip                       4.18.1           py36h6538335_0    conda-forge
six                       1.11.0                py36_1001    conda-forge
spefile                   1.6                        py_1    conda-forge
tifffile                  0.15.1          py36h452e1ab_1001    conda-forge
tk                        8.6.8                    vc14_0  [vc14]  conda-forge
tomopy                    1.2.1                    py36_0    conda-forge
toolz                     0.9.0                      py_1    conda-forge
tornado                   5.1.1           py36hfa6e2cd_1000    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.15.26706          h3a45250_0
wheel                     0.32.2                   py36_0
wincertstore              0.2              py36h7fe50ca_0
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```

I suspect it's a compilation issue between ASTRA and tomopy. I installed both via the ``conda install -n envir -c channel  package``-command in the environment, first tomopy and than astra-toolbox. 
Before, I somehow (not quite sure what changed in the meantime) got the file running in spyder but the astra reconstruction was just filled with 1e-6 values. If I run it via the terminal I get the above error. 
Does that mean that tomopy and astra call for different numpy packages? 

To me, it kind of seems similar to the #195 and #277  issue, but I'm using a windows machine (thats why I created an new issue, please merge, if it's the same problem). 
Has anyone encountered this behavior on windows? Do you think building from source would solve it for me, too? Shall I try reversing the installation order?
I'm a bit reluctant to try to build from source, because I haven't worked wit Visual studio and boost (have not installed that yet) before, but I'm willing to try if that would solve the issue. Maybe someone could guide me through this, so I'm not destroying something else in the process?

**Comment by dmpelt** [comment]
At first glance it seems to already fail during the import of tomopy, and actually while importing scikit-image.

Can you try to run a simple script that only does `import skimage`, and nothing else? I expect that to fail as well. 

The issue seems to be that some parts of your conda environment are build against one version of numpy, and other parts build against a different version. Looking at the environment, it see that `numpy` and `numpy-base` have different version numbers -- this could be related...

If the problem occurs during the skimage import, it is probably not a tomopy or astra problem, but hopefully we'll be able to help you fix it in any case!

**Comment by oachk** [comment]
Hi, 
thanks for the fast reply!
You're right, my `skimage` fails straight away.
I get the same error as above when I just run a simple `skimage` test script.
```
from skimage import data, io, filters

image = data.coins()
# ... or any other NumPy array!
edges = filters.sobel(image)
io.imshow(edges)
io.show()
```

**Comment by oachk** [comment]
I suspect that I broke some dependencies when I uninstalled astra-toolbox and reinstalled it, once I figured out that the reconstructions I git earlier were all empty. I'll give it a try in a new clean environment and see if I get my earlier problem with the empty reconstructions back

**Comment by oachk** [comment]
ok, I've created a new environment and only installed tomopy and astra toolbox using the following commands:
```
conda create --name tomoastra2 python=3.6
conda install -n tomoastra2 -c conda-forge tomopy
conda install -n tomoastra2 -c astra-toolbox astra-toolbox
```
If I list the packages I still get different numpy and numpy-base:
```
conda list
# packages in environment at C:\Users\Hanna\Miniconda3\envs\tomoastra2:
#
# Name                    Version                   Build  Channel
astra-toolbox             1.8.3            py36h8016247_0    astra-toolbox
blas                      1.0                         mkl
ca-certificates           2018.10.15           ha4d7672_0    conda-forge
certifi                   2018.10.15            py36_1000    conda-forge
cftime                    1.0.2.1         py36h452e1ab_1000    conda-forge
cloudpickle               0.6.1                      py_0    conda-forge
curl                      7.49.1                   vc14_2  [vc14]  conda-forge
cycler                    0.10.0                     py_1    conda-forge
cytoolz                   0.9.0.1         py36hfa6e2cd_1001    conda-forge
dask-core                 0.20.0                     py_0    conda-forge
decorator                 4.3.0                      py_0    conda-forge
dxchange                  0.1.4                    py36_0    conda-forge
dxfile                    0.5                      py36_0    conda-forge
edffile                   5.0.0                    py36_0    conda-forge
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
h5py                      2.8.0            py36hfa6e2cd_0    conda-forge
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
imageio                   2.4.1                 py36_1000    conda-forge
intel-openmp              2019.0                      118
jpeg                      9b                       vc14_2  [vc14]  conda-forge
kiwisolver                1.0.1           py36he980bc4_1002    conda-forge
libastra                  1.8.3                h49425a5_0    astra-toolbox
libnetcdf                 4.6.1                    vc14_2  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
matplotlib                2.2.2                    py36_1    conda-forge
mkl                       2019.0                      118
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
netcdf4                   1.4.0                    py36_0    conda-forge
networkx                  2.2                        py_1    conda-forge
numexpr                   2.6.8            py36h9ef55f4_0
numpy                     1.13.3           py36h5c71026_4
numpy-base                1.15.4           py36h8128ebf_0
olefile                   0.46                       py_0    conda-forge
openssl                   1.0.2p               hfa6e2cd_0
pillow                    5.2.0                    py36_0    conda-forge
pip                       18.1                     py36_0
pyparsing                 2.3.0                      py_0    conda-forge
pyqt                      5.6.0            py36h764d66f_7    conda-forge
pyreadline                2.1                   py36_1000    conda-forge
python                    3.6.6                he025d50_0    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pytz                      2018.7                     py_0    conda-forge
pywavelets                1.0.1           py36h452e1ab_1000    conda-forge
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
scikit-image              0.14.1          py36h6538335_1001    conda-forge
scipy                     1.1.0            py36hc28095f_0
setuptools                40.5.0                   py36_0
sip                       4.18.1           py36h6538335_0    conda-forge
six                       1.11.0                py36_1001    conda-forge
spefile                   1.6                        py_1    conda-forge
tifffile                  0.15.1          py36h452e1ab_1001    conda-forge
tk                        8.6.8                    vc14_0  [vc14]  conda-forge
tomopy                    1.2.1                    py36_0    conda-forge
toolz                     0.9.0                      py_1    conda-forge
tornado                   5.1.1           py36hfa6e2cd_1000    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.15.26706          h3a45250_0
wheel                     0.32.2                   py36_0
wincertstore              0.2              py36h7fe50ca_0
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```

And I also still get the skimage error:
```
(tomoastra2) C:\Users\Hanna>python .\Documents\PythonScripts\main_test_tomopy_astra.py
Traceback (most recent call last):
  File ".\Documents\PythonScripts\main_test_tomopy_astra.py", line 7, in <module>
    import tomopy
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\tomopy\__init__.py", line 57, in <module>
    from tomopy.misc.phantom import *
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\tomopy\misc\phantom.py", line 57, in <module>
    import skimage
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\__init__.py", line 177, in <module>
    from .data import data_dir
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\data\__init__.py", line 15, in <module>
    from ..io import imread, use_plugin
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\io\__init__.py", line 7, in <module>
    from .manage_plugins import *
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\io\manage_plugins.py", line 28, in <module>
    from .collection import imread_collection_wrapper
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\io\collection.py", line 14, in <module>
    from ..external.tifffile import TiffFile
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\external\tifffile\__init__.py", line 1, in <module>
    from .tifffile import imsave, imread, imshow, TiffFile, TiffWriter, TiffSequence
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra2\lib\site-packages\skimage\external\tifffile\tifffile.py", line 293, in <module>
    from . import _tifffile
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
```
Why does this happen if I'm not even working in the base environment, but in my new environment?

**Comment by dmpelt** [comment]
Can you try (in a new environment) to put the tomopy and Astra install commands in the same line?

conda install -n tomoastra3 -c conda-forge -c astra-toolbox astra-toolbox tomopy

This should help conda figure out the depencies, or give an error if it can't figure it out.

**Comment by oachk** [comment]
Hi, 
tried it, the installation worked without a problem.
But I still get the same error:

```
(tomoastra3) C:\Users\Hanna>python .\Documents\PythonScripts\main_test_tomopy_astra.py
Traceback (most recent call last):
  File ".\Documents\PythonScripts\main_test_tomopy_astra.py", line 7, in <module>
    import tomopy
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\tomopy\__init__.py", line 57, in <module>
    from tomopy.misc.phantom import *
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\tomopy\misc\phantom.py", line 57, in <module>
    import skimage
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\__init__.py", line 177, in <module>
    from .data import data_dir
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\data\__init__.py", line 15, in <module>
    from ..io import imread, use_plugin
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\io\__init__.py", line 7, in <module>
    from .manage_plugins import *
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\io\manage_plugins.py", line 28, in <module>
    from .collection import imread_collection_wrapper
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\io\collection.py", line 14, in <module>
    from ..external.tifffile import TiffFile
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\external\tifffile\__init__.py", line 1, in <module>
    from .tifffile import imsave, imread, imshow, TiffFile, TiffWriter, TiffSequence
  File "C:\Users\Hanna\Miniconda3\envs\tomoastra3\lib\site-packages\skimage\external\tifffile\tifffile.py", line 293, in <module>
    from . import _tifffile
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
```

**Comment by dmpelt** [comment]
Can you try to force the numpy version to 1.15 while installing tomopy and astra (again, in a new environment is safest), i.e.:

```
conda install -n tomoastra4 -c conda-forge -c astra-toolbox astra-toolbox tomopy numpy=1.15
```

There is some package that downgrades numpy to 1.13 while some other package probably expects 1.15, which could be the cause of the problems.

If this doesn't work, the other way around might work instead (i.e. forcing numpy to 1.13):

```
conda install -n tomoastra5 -c conda-forge -c astra-toolbox astra-toolbox tomopy numpy=1.13
```

This, at least, lets the other packages know that you will have numpy 1.13 in the end...

**Comment by oachk** [comment]
Hi again,
I also tried a new environment with just tomopy (no astra-toolbox) and the basic full-tomopy projection/reconstruction example from the documentation worked. But as soon as I added the astra-toolbox installation I run into the same error as above, without even calling astra. Really looks like the installation with astra-toolbox causes the trouble. 
I'm also still confused that I got it "running" with a suppressed error (and thus 'only' empty reconstructions) yesterday and now it already fails at the import statement? But maybe that's not so important.

Is there anything else I can test? If it's really a genuine conflict between tomopy and astra-toolbox requirements, other people should run into the same problem, which isn't happening, right? So it must be something with my python installation. Maybe I'm missing something obvious somewhere else...
Could it be that I have a conflicting package/python somewhere else on my pc?
Thanks anyway for helping me with this, I really appreciate it!

**Comment by dgursoy** [comment]
Can you give it a try with Python=3.5?

**Comment by carterbox** [comment]
@dgursoy There is no conda-forge windows build for python=3.5. because conda-forge moved their builds from python=3.5 to python=3.6 before we starting building on Windows.

**Comment by dgursoy** [comment]
I just realized that they stopped building for Python 3.5 starting from Oct 1. 

@dmpelt Probably unrelated but what is libastra for and if it is different than astra-toolbox (both share the same versions)? I checked if it is in the astra-toolbox channel, but I couldn't locate it.

@oachk I would try to install tomopy and astra-toolbox separately in different environments to see which numpy version each one prefers.

**Comment by oachk** [comment]
Hi!
Sorry for the late reply. I did as you suggested @dgursoy and create two seperate environments. See the package lists below. As expected tomopy installs numpy 1.15, while astra-toolbox wants numpy 1.13. I used the channels conda-forge for tomopy and astra-toolbox for astra-toolbox and installed nothing else. They apparently also use a slightly different python (3.6.6 and 3.6.7). Not sure what changed between these version and if that can have an influence, too?
```
(onlytomopy) C:\Users\Hanna>conda list
# packages in environment at C:\Users\Hanna\Miniconda3\envs\onlytomopy:
#
# Name                    Version                   Build  Channel
blas                      1.0                         mkl
ca-certificates           2018.10.15           ha4d7672_0    conda-forge
certifi                   2018.10.15            py36_1000    conda-forge
cftime                    1.0.2.1         py36h452e1ab_1000    conda-forge
cloudpickle               0.6.1                      py_0    conda-forge
curl                      7.49.1                   vc14_2  [vc14]  conda-forge
cycler                    0.10.0                     py_1    conda-forge
cytoolz                   0.9.0.1         py36hfa6e2cd_1001    conda-forge
dask-core                 0.20.1                     py_0    conda-forge
decorator                 4.3.0                      py_0    conda-forge
dxchange                  0.1.4                    py36_0    conda-forge
dxfile                    0.5                      py36_0    conda-forge
edffile                   5.0.0                    py36_0    conda-forge
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
h5py                      2.8.0            py36hfa6e2cd_0    conda-forge
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
imageio                   2.4.1                 py36_1000    conda-forge
intel-openmp              2019.0                      118
jpeg                      9b                       vc14_2  [vc14]  conda-forge
kiwisolver                1.0.1           py36he980bc4_1002    conda-forge
libnetcdf                 4.6.1                    vc14_2  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
matplotlib                2.2.2                    py36_1    conda-forge
mkl                       2019.0                      118
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
netcdf4                   1.4.0                    py36_0    conda-forge
networkx                  2.2                        py_1    conda-forge
numexpr                   2.6.8           py36h6538335_1000    conda-forge
numpy                     1.15.4           py36ha559c80_0
numpy-base                1.15.4           py36h8128ebf_0
olefile                   0.46                       py_0    conda-forge
openssl                   1.0.2o                   vc14_0  [vc14]  conda-forge
pillow                    5.2.0                    py36_0    conda-forge
pip                       18.1                  py36_1000    conda-forge
pyparsing                 2.3.0                      py_0    conda-forge
pyqt                      5.6.0            py36h764d66f_7    conda-forge
pyreadline                2.1                   py36_1000    conda-forge
python                    3.6.6                he025d50_0    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pytz                      2018.7                     py_0    conda-forge
pywavelets                1.0.1           py36h452e1ab_1000    conda-forge
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
scikit-image              0.14.1          py36h6538335_1001    conda-forge
scipy                     1.1.0            py36h4f6bf74_1
setuptools                40.5.0                   py36_0    conda-forge
sip                       4.18.1           py36h6538335_0    conda-forge
six                       1.11.0                py36_1001    conda-forge
spefile                   1.6                        py_1    conda-forge
tifffile                  0.15.1          py36h452e1ab_1001    conda-forge
tk                        8.6.8                    vc14_0  [vc14]  conda-forge
tomopy                    1.2.1                    py36_0    conda-forge
toolz                     0.9.0                      py_1    conda-forge
tornado                   5.1.1           py36hfa6e2cd_1000    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.0.25420                    0    conda-forge
wheel                     0.32.2                   py36_0    conda-forge
wincertstore              0.2                   py36_1002    conda-forge
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```
and 
```
(onlyastra) C:\Users\Hanna>conda list
# packages in environment at C:\Users\Hanna\Miniconda3\envs\onlyastra:
#
# Name                    Version                   Build  Channel
astra-toolbox             1.8.3            py36h8016247_0    astra-toolbox
blas                      1.0                         mkl
certifi                   2018.10.15               py36_0
icc_rt                    2017.0.4             h97af966_0
intel-openmp              2019.0                      118
libastra                  1.8.3                h49425a5_0    astra-toolbox
mkl                       2019.0                      118
mkl_fft                   1.0.6            py36hdbbee80_0
mkl_random                1.0.1            py36h77b88f5_1
numpy                     1.13.3           py36h5c71026_4
pip                       18.1                     py36_0
python                    3.6.7                h33f27b4_1
scipy                     1.1.0            py36hc28095f_0
setuptools                40.5.0                   py36_0
six                       1.11.0                   py36_1
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.15.26706          h3a45250_0
wheel                     0.32.2                   py36_0
wincertstore              0.2              py36h7fe50ca_0
```

**Comment by carterbox** [comment]
This is silly. The error occurs during the import of `scikit-image`. We know this because that's what the error message is. Apparently something is wrong with importing this version of skimage.
```
scikit-image              0.14.1          py36h6538335_1001    conda-forge
```
I was able to reproduce your problem on a Windows machine. Pinning scikit-image to version 0.14.0 solved the problem with the import error.

**Comment by AnantMitra** [comment]
I am still facing the same problem: 
Install Astra is downgrading the following packages:

The following packages will be SUPERSEDED by a higher-priority channel:

  mkl_fft            conda-forge::mkl_fft-1.3.0-py36h6434a~ --> main::mkl_fft-1.0.15-py36h14836fe_0
  numexpr            conda-forge::numexpr-2.7.3-py36he38d9~ --> main::numexpr-2.7.0-py36h6288b17_0
  numpy              conda-forge::numpy-1.19.5-py36h4b40d7~ --> main::numpy-1.13.3-py36hb69e940_3
  scikit-image       conda-forge::scikit-image-0.17.2-py36~ --> main::scikit-image-0.15.0-py36ha925a31_0
  scipy              conda-forge::scipy-1.5.3-py36h7ff6e69~ --> main::scipy-1.3.2-py36h29ff71c_0

Is there any way to resolve this?

P.S. Already tried "conda install -n myenv -c conda-forge -c astra-toolbox astra-toolbox tomopy scikit-image=0.14.0"  but it didn't actually installed 'Astra-toolbox'.

**Comment by dmpelt** [comment]
You could try to install a development build of astra by using '-c astra-toolbox/label/dev' instead of '-c astra-toolbox'. Not sure if this will fix the problem, but there might be a better chance of astra being compiled with more recent versions of other packages in this case.
